pkg/daemon/daemon.go

```diff
@@ -466,7 +466,8 @@ func ReexecuteForTargetRoot(target string) error {
 			// Otherwise, we assume that there's no suffixing needed.  Hopefully
 			// by RHEL10 the MCD will have fundamentally changed and we won't be doing the
 			// chroot() thing anymore.
-			klog.Info("not chrooting for source=rhel-%s target=rhel-%s", sourceMajor, targetMajor)
+			klog.Infof("not chrooting for source=rhel-%s target=rhel-%s", sourceMajor, targetMajor)
```

Comment:
Actually sorry, hold on a second...I think I was wrong in our chat.  We do need to chroot right?  We just don't want to re-exec...

Something more like

```
diff --git a/pkg/daemon/daemon.go b/pkg/daemon/daemon.go
index 6a9f2079c..2e18bf5da 100644
--- a/pkg/daemon/daemon.go
+++ b/pkg/daemon/daemon.go
@@ -471,35 +471,38 @@ func ReexecuteForTargetRoot(target string) error {
 	} else {
 		klog.Info("assuming we can use container binary chroot() to host")
 	}
-	sourceBinary := "/usr/bin/machine-config-daemon" + sourceBinarySuffix
-	src, err := os.Open(sourceBinary)
-	if err != nil {
-		return fmt.Errorf("opening %s: %w", sourceBinary, err)
-	}
-	defer src.Close()
+	var targetBin string
+	if sourceBinarySuffix != "" {
+		sourceBinary := "/usr/bin/machine-config-daemon" + sourceBinarySuffix
+		src, err := os.Open(sourceBinary)
+		if err != nil {
+			return fmt.Errorf("opening %s: %w", sourceBinary, err)
+		}
+		defer src.Close()
 
-	targetBinBase := "run/bin/machine-config-daemon"
-	targetBin := filepath.Join(target, targetBinBase)
-	targetBinDir := filepath.Dir(targetBin)
-	if _, err := os.Stat(targetBinDir); err != nil {
-		if err := os.Mkdir(targetBinDir, 0o755); err != nil {
-			return fmt.Errorf("mkdir %s: %w", targetBinDir, err)
+		targetBinBase := "run/bin/machine-config-daemon"
+		targetBin = filepath.Join(target, targetBinBase)
+		targetBinDir := filepath.Dir(targetBin)
+		if _, err := os.Stat(targetBinDir); err != nil {
+			if err := os.Mkdir(targetBinDir, 0o755); err != nil {
+				return fmt.Errorf("mkdir %s: %w", targetBinDir, err)
+			}
 		}
-	}
 
-	f, err := os.Create(targetBin)
-	if err != nil {
-		return fmt.Errorf("writing %s: %w", targetBin, err)
-	}
-	if _, err := io.Copy(f, src); err != nil {
+		f, err := os.Create(targetBin)
+		if err != nil {
+			return fmt.Errorf("writing %s: %w", targetBin, err)
+		}
+		if _, err := io.Copy(f, src); err != nil {
+			f.Close()
+			return fmt.Errorf("writing %s: %w", targetBin, err)
+		}
+		if err := f.Chmod(0o755); err != nil {
+			return err
+		}
+		// Must close our writable fd
 		f.Close()
-		return fmt.Errorf("writing %s: %w", targetBin, err)
 	}
-	if err := f.Chmod(0o755); err != nil {
-		return err
-	}
-	// Must close our writable fd
-	f.Close()
 
 	if err := syscall.Chroot(target); err != nil {
 		return fmt.Errorf("failed to chroot to %s: %w", target, err)
@@ -509,6 +512,9 @@ func ReexecuteForTargetRoot(target string) error {
 		return fmt.Errorf("failed to change directory to /: %w", err)
 	}
 
+	if targetBin == "" {
+		return nil
+	}
 	// Now we will see the binary in the target root
 	targetBin = "/" + targetBinBase
 	// We have a "belt and suspenders" approach for detecting the case where
```

?

But there's still something weird going on here because I don't understand how we can be recursing here...IOW why are we getting *that* error of having the text file being busy?

Reply:
Since, you already have the changes, can you push them to this PR? This doesn't apply cleanly for me